src/lib.rs

```diff
@@ -837,7 +863,11 @@ impl<'a, T: Debug + One + PrimInt + Zero> Iterator for IterSetBits<'a, T> {
                         // There is another bit set after i in the block.
                         let bs = b * bits_per_block::<T>() + i_off + tz;
                         self.range.start = bs + 1;
-                        return Some(bs);
+                        if bs >= self.range.end {
```

Comment:
Perhaps a separate issue, but the names `bs` and `tz` are not very descriptive. I'm thinking of timezones (`tz`) and, well `bs`, use your imagination :)

Closer inspection: `tz` is trailing zeroes,  but `bs` remains difficult to understand. What does it stand for?

I know I've reviewed this before, but as time passes I forget.

Reply:
`b` is also not a good name. It could be `bits`, `bytes`, `blocks`...

In this instance I think it's a block.

Reply:
Whether they're good names or not, I don't think this PR is the right place to change them: it would lead to a lot of churn that isn't related to the PR. [FWIW, I found reading this bit of code fairly easy, and I can't say that of every bit of code I've written!]

Reply:
`bs` = "bit set position"

`tz` = "trailing zeros" (given that the line is `let tz = (v >> i_off).trailing_zeros() as usize;` I think this one is pretty clear ;) )

Reply:
As mentioned, I agree that this isn't the place for variable name changes above. But on this subject, (and probably a personal preference mismatch) I like variable names that don't force me to read too much surrounding code. If the variable name is obvious at a glance, then I can comprehend it quicker. I'd never have guessed  "bit set position" from `bs`, but `bit_pos` would have been excellent, and this conversation wouldn't be happening ;)

Anyway, resuming review.

Reply:
I'm trying to understand why we have ` + tz` in `let bs = b * bits_per_block::<T>() + i_off + tz;`.

I think I'm confused because I thought the bits would be stored "in order" even between blocks, e.g. if we have 8-bit blocks, the idices: `1 2 3 4 5 6 7 8 | 9 10 11 12...`, but I think they are really stored `8 7 6 5 4 3 2 1 | 16 15 14...`, which is why you have to skip over trailing zeroes, right?

Reply:
Your best bet is to run the code and put in print statements -- it is hard to reason about it statically if you're unfamiliar with the various tricks used here for performance. That said, the comment hopefully gives a good indication: `Find the next bit set after 'i % bits_per_block()'`. You'll also need to read up on `trailing_zeros`.

Reply:
[I can try an English explanation, though I don't know if it helps. "i_off is the last known set bit. tz_off is the number of unset bits to the next set bit. Add that to the overall block offset."]

Reply:
OK, i misunderstood the meaning of `tz`. I thought it was the number of unallocated trailing zeros at the end of the block.

I'm thinking I'm doing a too in-depth review here. Your changes look fine.